en/docs/concepts/oidc/hybrid-client-profile.md

```diff
@@ -0,0 +1,242 @@
+ “exp”: 1510831010,
+ “nonce”: “asd”,
+ “iat”: 1510831007
+}
+```
+
+There can be instances where you get two access tokens. One from the
+authorization endpoint and the other from the token endpoint. These two
+access tokens may or may not be same.
+
+### code id_token 
+
+This ` response_type ` requests a code and an id\_token
+from the authorization endpoint.
+
+Following is a sample authorization request that uses
+` code id_token ` as the ` response_type `
+:
+
+``` java
+https://localhost:9443/oauth2/authorize?response_type=code id_token&client_id=<Client ID>&nonce=asd&redirect_uri=http://localhost:8080/playground2/oauth2client&scope=openid
+```
+
+Following is the sample response that you will receive on successful
+authorization:
+
+``` java
+http://localhost:8080/playground2/oauth2client?code=16fd899f-5f0c-3114-875e-2547b629cd05&id_token=eyJ4NXQiOiJOVEF4Wm1NeE5ETXlaRGczTVRVMVpHTTBNekV6T0RKaFpXSTRORE5sWkRVMU9HRmtOakZpTVEiLCJraWQiOiJOVEF4Wm1NeE5ETXlaRGczTVRVMVpHTTBNekV6T0RKaFpXSTRORE5sWkRVMU9HRmtOakZpTVEiLCJhbGciOiJSUzI1NiJ9.eyJhdF9oYXNoIjoiSnJaWTlNdFlWRUlJSlV4LUREQm13dyIsImNfaGFzaCI6IlM1VU9YUk5OeVlzSTZaMEczeHhkcHciLCJzdWIiOiJhZG1pbiIsImF1ZCI6WyJuY3piZzVtNXh4dDZ0UDRVTVp3QjZQdFFvUW9hIl0sImF6cCI6Im5jemJnNW01eHh0NnRQNFVNWndCNlB0UW9Rb2EiLCJpc3MiOiJodHRwczpcL1wvbG9jYWxob3N0Ojk0NDNcL29hdXRoMlwvdG9rZW4iLCJleHAiOjE1MTA4MzE1MTIsIm5vbmNlIjoiYXNkIiwiaWF0IjoxNTEwODMxNTA4fQ.BsiXZwP_EFnNH-5r01z4P18OZbVY1WHOD1GSTrDa4-TxcSEuMOlvIQA54Poy0hUS8RCP46XB-WhUaOHQpvsHBj6CUCkNWAqJj5F-TetXUhONhnI0Hp7K3zofa_E5-ucFmUoKVwk-wFAMakKziIsX9P8v9-mi2kPlQPDyS3i7tkRlABS5emgbOSHxNsoKjdaglLT78zdARMFfF0i0oaDyRv9nfZIgSZJE1Qec99DA7engA43NJQCB1vMjF9Qruefyyjtq2abaLLRG6Yh6NeWDyIXkjjbHEcKxzBsKU6VqL84DqHHYFUwZ1nL2aLon1kHXUHgGfuhuBJ5qIwEtbZrQLw#session_state=d96bad64e37e82196898a824082aafbdd945c922e7d40cb4e0013d9fad6d68c8.o0_m4GJ1YJvNUUqg8k3LrQ
+```
+
+When you decrypt the ` id_token ` it will be as follows:
+
+``` java
+{
+ “at_hash”: “JrZY9MtYVEIIJUx-DDBmww”,
+ “c_hash”: “S5UOXRNNyYsI6Z0G3xxdpw”,
+ “sub”: “admin”,
+ “aud”: [
+ “nczbg5m5xxt6tP4UMZwB6PtQoQoa”
+ ],
+ “azp”: “nczbg5m5xxt6tP4UMZwB6PtQoQoa”,
+ “iss”: “https://localhost:9443/oauth2/token",
+ “exp”: 1510831512,
+ “nonce”: “asd”,
+ “iat”: 1510831508
+}
+```
+
+!!! tip 
+    <p>Here, the id\_token is required to have a c\_hash value.</p>
+ 
+    <p>c\_hash is the base64url encoding of the left-most half of a hash of the
+    octets in the ASCII representation of a code value, where the hash
+    algorithm used is the hash algorithm of the ` alg `
+    header parameter of the ID token’s JOSE header.</p>
+ 
+    <p>The c\_hash value is mandatory when an ` id_token ` is
+    issued with code, and the ` response_type ` is equal to
+    ` code id_token ` or
+    ` code id_token token ` .</p>
+ 
+    <p>You can send the code to the token endpoint to request for an access
+    token, refresh token and id\_token. For this you can use the same curl
+    command provided for the code token specified
+    [above](#curl)."</p>
```

Comment:
To request for an access token, refresh token, and id_token, send the code to the token endpoint. You can use the same [curl command](#curl) that was provided for the code id_token.